### ISSUE TITLE
Error: ws does not work in the browser. Browser clients must use the native WebSocket object

### ISSUE THREAD

**Issue by Jaykingamez** [aws/aws-iot-device-sdk-js-v2#206]
Confirm by changing [ ] to [x] below to ensure that it's a bug:
- [x] I've searched for [previous similar issues](https://github.com/aws/aws-iot-device-sdk-js-v2/issues) and didn't find any solution

Code like [this](https://github.com/aws/aws-iot-device-sdk-js-v2/issues/77) brings about 

> Error: ws does not work in the browser. Browser clients must use the native WebSocket object

```
ngOnInit() {
    const clientBootstrap = new io.ClientBootstrap();

    const myClient = iot.AwsIotMqttConnectionConfigBuilder
      .new_with_websockets({
        region: 'ap-southeast-1',
        credentials_provider: undefined,
        proxy_options: undefined
      });

    const config = myClient.with_client_id('wsssss')
      .with_clean_session(true)
      .with_credentials('ap-southeast-1', 'xxxx', 'xxxx', '')
      .with_endpoint('xxxx-ats.iot.ap-southeast-1.amazonaws.com')
      .build();

    const client = new mqtt.MqttClient(clientBootstrap);

    const connection = client.new_connection(config);

    connection.on('connect', (session_present) => {
      console.log(session_present);
      console.log('connected');
      connection.subscribe('/test/me/senpai', mqtt.QoS.AtLeastOnce, (topic, payload) => {
        const decoder = new TextDecoder('utf8');
        const message = decoder.decode(payload);
        console.log(`message received: topic=${topic} message=${message}`);
      }).then(subscription => {
        return connection.publish(subscription.topic, 'NOTICE ME', subscription.qos);
      });
    });

    connection.connect();
  }
```

**SDK version number**
1.6.4

**Expected behavior**
For the error to not pop up and for the code to function as usual

**Additional context**
A temporary fix would be rolling back to version 1.6.3 where it was still working

**Comment by bretambrose** [comment]
Well that doesn't inspire confidence in the browser CI.  I'll take a look at isomorphic-js as a polyfill/replacement.

**Comment by omar-enrique** [comment]
@bretambrose We are having the same issue with our app that uses AWS IOT.  I actually came here to report this and saw this issue.  It happens for us on the `const connection = client.new_connection(config)`.  What was weird is that we were still in SDK version `1.6.2` when we ran into the issue, it probably has more to do with the MQTT submodule than it does IOT Core's SDK

**Comment by bretambrose** [comment]
This should be fixed in 1.6.5
